### PR TITLE
improve zh_cn translation

### PR DESCRIPTION
**Pull request by KevinTong9**
- [ ] The changes have been tested locally
- [x] There are no breaking changes

* **What kind of change does this PR introduce?**
ZH_CN translation update

* **What is the current behavior?**
[Pinecil V1 small display error in ZH_CN](https://github.com/Ralim/IronOS/issues/1780)

* **Other information**:
It doesn't aim to fix the problem in the issue, just a translation update.

### CODE REVIEW

File: Translations/translation_ZH_CN.json [hunk 1]
```diff
@@ -46,7 +46,7 @@
       "message": "电压过低"
     },
     "UndervoltageString": {
-      "message": "Undervoltage\n"
+      "message": "欠压\n"
```
Comment (discip) [inline Translations/translation_ZH_CN.json:49]:
```suggestion
      "message": "欠压"
```

File: Translations/translation_ZH_CN.json [hunk 2]
```diff
@@ -107,16 +107,16 @@
   },
   "menuValues": {
     "USBPDModeDefault": {
-      "displayText": "Default\nMode"
+      "displayText": "默认\n模式"
```
Comment (discip) [inline Translations/translation_ZH_CN.json:110]:
```suggestion
      "displayText": "默认模式"
```

File: Translations/translation_ZH_CN.json [hunk 3]
```diff
@@ -107,16 +107,16 @@
...
     },
     "USBPDModeNoDynamic": {
       "displayText": "No\nDynamic"
     },
     "USBPDModeSafe": {
-      "displayText": "Safe\nMode"
+      "displayText": "安全\n模式"
```
Comment (discip) [inline Translations/translation_ZH_CN.json:116]:
```suggestion
      "displayText": "安全模式"
```

File: Translations/translation_ZH_CN.json [hunk 4]
```diff
@@ -107,16 +107,16 @@
...
     },
     "TipTypeAuto": {
-      "displayText": "Auto\nSense"
+      "displayText": "自动\n检测"
```
Comment (discip) [inline Translations/translation_ZH_CN.json:119]:
```suggestion
      "displayText": "自动检测"
```

File: Translations/translation_ZH_CN.json [hunk 5]
```diff
@@ -252,8 +252,8 @@
       "description": "霍尔效应传感器用作启动待机模式的灵敏度 <1=最低灵敏度 | ... | 9=最高灵敏度>"
     },
     "HallEffSleepTimeout": {
-      "displayText": "HallSensor\nSleepTime",
-      "description": "Interval before \"sleep mode\" starts when hall effect is above threshold"
+      "displayText": "霍尔传感器\n休眠时间",
```
Comment (discip) [inline Translations/translation_ZH_CN.json:255]:
```suggestion
      "displayText": "霍尔传感器休眠时间",
```

File: Translations/translation_ZH_CN.json [hunk 6]
```diff
@@ -340,8 +340,8 @@
       "description": ""
     },
     "SolderingTipType": {
-      "displayText": "Soldering\nTip Type",
-      "description": "Select the tip type fitted"
+      "displayText": "焊接头\n类型",
```
Comment (discip) [inline Translations/translation_ZH_CN.json:343]:
```suggestion
      "displayText": "焊接头类型",
```

**Review by discip — CHANGES_REQUESTED** [review]
@KevinTong9
Have you tried the changes on your device?
If I remember correctly, you would need to remove all the new lines **`\n`** in order to make it work.

**Comment by KevinTong9** [comment]
> @KevinTong9 Have you tried the changes on your device? If I remember correctly, you would need to remove all the new lines **`\n`** in order to make it work.

Sorry, I haven't tested it locally as my current environment doesn't allow it. Testing might have to wait until I have some free time.

By the way, I can only test Pencil V1.

**Comment by KevinTong9** [comment]
> @KevinTong9 Have you tried the changes on your device? If I remember correctly, you would need to remove all the new lines **`\n`** in order to make it work.

**`\n`** deleted now

**Review by discip — APPROVED** [review]

**Comment by discip** [comment]
I think it is best if you test the changes prior to merging.

I could do the testing for you as soon as I'm home again.

**Comment by discip** [comment]
I think it is best if we test the changes prior to merging.

I could do the testing for you as soon as I'm home again.

**Comment by KevinTong9** [comment]
> I think it is best if we test the changes prior to merging.
> 
> I could do the testing for you as soon as I'm home again.

Tk. If I have time tonight, I can also go test V1.

**Comment by KevinTong9** [comment]
> I did test it, but in some cases I was not sure if you would be satisfied. SO from my point of view this should be good to go, but please take your time testing it yourself. 😊
> 
> kind regards

The translation works fine for me.  This looks good to go.

*That said, it did not address the problem I mentioned in the issue, which is unrelated to this PR. The issue of displaying garbled text still persists. Perhaps you could look into the root of the issue？It seems like the characters were not displayed completely. It should display "欠压" here, but with only the upper half visible now.*



Best regards

**Comment by discip** [comment]
> That said, it did not address the problem I mentioned in the issue ...

I'm pretty sure @Ralim could fix this, but I don't know if he has got time to do that. 😊

**Comment by ia** [comment]
> _That said, it did not address the problem I mentioned in the issue, which is unrelated to this PR. The issue of displaying garbled text still persists. Perhaps you could look into the root of the issue？It seems like the characters were not displayed completely. It should display "欠压" here, but with only the upper half visible now._
> 
> 

Hello. Could you tell, please, what kind of power supply do you use so I could try to reproduce the issue? My assumption is that it's something like more or less standard 5V/1A old USB power brick but I just would like to make sure. Thanks.

P.S. Can't promise anything but will try to reproduce and think about it. _I may have_ a couple of ideas to check.

**Comment by ia** [comment]
> Hello. Could you tell, please, what kind of power supply do you use so I could try to reproduce the issue? My assumption is that it's something like more or less standard 5V/1A old USB power brick but I just would like to make sure. Thanks.

And which power port do you use? DC 5525 barrel or USB-C? And what is on the other side of the cord with the iron exactly?

I tried to simulate the problem using _TS80P_ with different static sets of voltage (everything I have closer around _~5V_): with _3V3_ it doesn't turn on, with _4.5V_ it works, and even trying to heat the tip, lol.

Anyway, since @Ralim did put `\n` with [this commit](https://github.com/Ralim/IronOS/pull/2013/files), I would try to revert `\n` to the `UndervoltageString` string and to see if there is any difference.

**Comment by KevinTong9** [comment]
> > _That said, it did not address the problem I mentioned in the issue, which is unrelated to this PR. The issue of displaying garbled text still persists. Perhaps you could look into the root of the issue？It seems like the characters were not displayed completely. It should display "欠压" here, but with only the upper half visible now._
> > 
> 
> Hello. Could you tell, please, what kind of power supply do you use so I could try to reproduce the issue? My assumption is that it's something like more or less standard 5V/1A old USB power brick but I just would like to make sure. Thanks.
> 
> P.S. Can't promise anything but will try to reproduce and think about it. _I may have_ a couple of ideas to check.

Already: [Pinecil V1 small display error in ZH_CN](https://github.com/Ralim/IronOS/issues/1780)

- USB-C
- 5V
- Any brick will do, computer port could too.
- **Pencil V1 with B2 tip** not TS80P

**Comment by KevinTong9** [comment]
> > That said, it did not address the problem I mentioned in the issue ...
> 
> I'm pretty sure @Ralim could fix this, but I don't know if he has got time to do that. 😊

Take your time. No rush at all—whenever it’s convenient for you.

If the translation looks good, this PR should be ready to merge.

Really appreciate it!

**Comment by discip** [comment]
@ia
> Anyway, since @Ralim did put `\n` with [this commit](https://github.com/Ralim/IronOS/pull/2013/files), I would try to revert `\n` to the `UndervoltageString` string and to see if there is any difference.

This won't help since the CJK languages ​​can only be drawn large.

The solution must be put in here somewhere: https://github.com/Ralim/IronOS/blob/dev/source/Core/Threads/UI/drawing/mono_96x16/draw_warning_undervoltage.cpp

I've also tested other languages ​​with the same result.

The solution would be to remove **`VIN:`** and draw the voltage on the right, but I'm not sure how to achieve the latter and unfortunately I don't have the time to get there empirically.

**Comment by discip** [comment]
Merging this in and the undervoltage issue should be addressed separately, since this affects other CJK languages as well.

**Comment by ia** [comment]
> This won't help since the CJK languages ​​can only be drawn large.

How interesting! Thank you for this piece of information and pardon me for misinformation. There are still so many nuances in the project which I'm just not fully aware yet still.

> The solution must be put in here somewhere: https://github.com/Ralim/IronOS/blob/dev/source/Core/Threads/UI/drawing/mono_96x16/draw_warning_undervoltage.cpp

Yeah, I was thinking about the same thing.

> I've also tested other languages ​​with the same result.

Aha, got it.

> The solution would be to remove **`VIN:`** and draw the voltage on the right, but I'm not sure how to achieve the latter and unfortunately I don't have the time to get there empirically.

I would like to help with this gladly, but unfortunately I have no idea how to reproduce the issue in the first place with my _TS80P_. I may have to get my hands on _Pinecil V2_ but I'm not sure if it will be reproducible there just like with _Pinecil V1_.

**Comment by ia** [comment]
> If the translation looks good, this PR should be ready to merge.

@KevinTong9, but just out of curiosity, could you tell, please: what about such lines as `USBPDModeNoDynamic`, `InputVoltageString` and types of tips? Did you miss them by accident or did you decide to keep them as universal non-translatable on purpose? Just asking, thanks.

**Pull request merged**